sd.py

```diff
@@ -54,16 +54,25 @@ def __init__(self, device, sd_version='2.1', hf_key=None):
             raise ValueError(f'Stable-diffusion version {self.sd_version} not supported.')
 
         # Create model
-        self.vae = AutoencoderKL.from_pretrained(model_key, subfolder="vae").to(self.device)
-        self.tokenizer = CLIPTokenizer.from_pretrained(model_key, subfolder="tokenizer")
-        self.text_encoder = CLIPTextModel.from_pretrained(model_key, subfolder="text_encoder").to(self.device)
-        self.unet = UNet2DConditionModel.from_pretrained(model_key, subfolder="unet").to(self.device)
+        if vram_O > 0:
+            pipe = StableDiffusionPipeline.from_pretrained(model_key, torch_dtype=torch.float16)
+            if vram_O > 1:
+                pipe.enable_sequential_cpu_offload()
+            pipe.enable_attention_slicing(1)
+            self.vae = pipe.vae
+            self.tokenizer = pipe.tokenizer
+            self.text_encoder = pipe.text_encoder
+            self.unet = pipe.unet
+            self.scheduler = DDIMScheduler.from_pretrained(model_key, subfolder="scheduler", torch_dtype=torch.float16)
+        else:
```

Comment:
we can remove the following lines and always use pipeline.

Reply:
:ok_hand: